File: api/src/endpoint/question/create_question.rs
use serde::{Deserialize, Serialize};
use api_derive::EndpointRequest;

use crate::{endpoint::Endpoint, resource::question::QuestionResource};

pub struct CreateQuestion;

impl Endpoint for CreateQuestion {
    const PATH: &'static str = "/repository/{owner}/{repo}/question";
    const METHOD: http::Method = http::Method::POST;

    type Request = CreateQuestionRequest;
    type Response = CreateQuestionResponse;
}

#[derive(EndpointRequest, Debug, Serialize, Deserialize)]
pub struct CreateQuestionRequest {
    pub title: String,
    pub body: String,
}

pub type CreateQuestionResponse = QuestionResource;

File: gitdot-api-ts/src/endpoint/question/create-question.ts
import { z } from "zod";

import { QuestionResource } from "../../resource";

export const CreateQuestionRequest = z.object({
  title: z.string(),
  body: z.string(),
});
export type CreateQuestionRequest = z.infer<typeof CreateQuestionRequest>;

export const CreateQuestion = {
  path: "/repository/{owner}/{repo}/question",
  method: "POST",
  request: CreateQuestionRequest,
  response: QuestionResource,
} as const;
export type CreateQuestion = typeof CreateQuestion;
